[linux-6.6.y] Driver for Zhaoxin SM3 SM4 algorithms

leoliu-oc opened the pull request:
Includes:
1. Add support for Zhaoxin GMI SM3 Secure Hash algorithm
This SM3 algorithm driver is developed to support the SM3 instruction,
making user develop their applications with both high performance and high
security.
2. Add support for Zhaoxin GMI SM4 Block Cipher algorithm
This SM4 algorithm driver is developed to support the SM4 instruction,
making user develop their applications with both high performance and high
security.

opsiff commented on arch/x86/crypto/sm3_zhaoxin_gmi.c:
```diff
@@ -0,0 +1,203 @@
+	if (gmi_available() == 0)
+		return crypto_register_shash(&zx_sm3_alg);
+
+	pr_warn("GMI is unavailable on this platform.");
```
这里应该是pr_info 不应该是pr_warn吧 所有x86都会走

leoliu-oc replied:
更新了一般，这里调整了逻辑。在匹配成功后才会pr_info打印。匹配不成功时不打印。这样就只有在兆芯平台才会有打印信息。

opsiff commented on arch/x86/crypto/sm4_zhaoxin_gmi.c:
```diff
@@ -0,0 +1,859 @@
+			pr_notice("GMI SM4 is detected by CPUID\n");
+			return 0;
+		}
+		pr_notice("GMI SM4 is available\n");
```
is available 为什么要用pr_notice?

leoliu-oc replied:
新版本中改为了pr_debug，供debug时判定用。正常情况下不输出log。

opsiff commented on arch/x86/crypto/sm4_zhaoxin_gmi.c:
```diff
@@ -0,0 +1,859 @@
+			if ((edx & 0x0030) != 0x0030)
+				return -ENODEV;
+
+			pr_notice("GMI SM4 is detected by CPUID\n");
```
is available 为什么要用pr_notice?

leoliu-oc replied:
同上。

opsiff commented on arch/x86/crypto/sm3_zhaoxin_gmi.c:
```diff
@@ -0,0 +1,203 @@
+			if ((edx & 0x0030) != 0x0030)
+				return -ENODEV;
+
+			pr_notice("GMI SM3 detected by CPUID\n");
```
is available 为什么要用pr_notice?

leoliu-oc replied:
同上。

opsiff commented on arch/x86/crypto/sm3_zhaoxin_gmi.c:
```diff
@@ -0,0 +1,203 @@
+			pr_notice("GMI SM3 detected by CPUID\n");
+			return 0;
+		}
+		pr_notice("GMI SM3 is available\n");
```
is available 为什么要用pr_notice?

leoliu-oc replied:
同上。

leoliu-oc committed:
crypto: Add support for Zhaoxin GMI SM4 Block Cipher algorithm

zhaoxin inclusion
category: feature
CVE: NA

-----------------

This SM4 algorithm driver is developed to support the SM4 instruction,
making user develop their applications with both high performance and high
security.

Signed-off-by: leoliu-oc <leoliu-oc@zhaoxin.com>

leoliu-oc committed:
crypto: Add support for Zhaoxin GMI SM3 Secure Hash algorithm

zhaoxin inclusion
category: feature
CVE: NA

-----------------

This SM3 algorithm driver is developed to support the SM3 instruction,
making user develop their applications with both high performance and high
security.

Signed-off-by: leoliu-oc <leoliu-oc@zhaoxin.com>

opsiff commented:
/lgtm

opsiff commented:
/ok-to-test

opsiff committed:
Merge branch 'linux-6.6.y' into linux-6.6.y-47-sm3-sm4
